app/upgrades/v15/upgrades.go

```diff
@@ -21,7 +27,34 @@ func CreateUpgradeHandler(
 			return vm, err
 		}
 
-		ctx.Logger().Info("Upgrade complete")
+		UpgradeSigningInfos(ctx, keepers.SlashingKeeper)
+
+		ctx.Logger().Info("Upgrade v15 complete")
 		return vm, err
 	}
 }
+
+// UpgradeSigningInfos updates the signing infos of validators for which
+// the consensus address is missing
+func UpgradeSigningInfos(ctx sdk.Context, sk slashingkeeper.Keeper) {
+	signingInfos := []slashingtypes.ValidatorSigningInfo{}
+
+	// update consensus address in signing info
+	// using the store key of validators
+	sk.IterateValidatorSigningInfos(ctx, func(address sdk.ConsAddress, info slashingtypes.ValidatorSigningInfo) (stop bool) {
+		if info.Address == "" {
+			info.Address = address.String()
+			signingInfos = append(signingInfos, info)
+		}
+
+		return false
+	})
+
+	for _, si := range signingInfos {
+		addr, err := sdk.ConsAddressFromBech32(si.Address)
+		if err != nil {
+			panic(err)
```

Comment:
Just to be safe, avoid the panic, log an error message and move to the next signing info.